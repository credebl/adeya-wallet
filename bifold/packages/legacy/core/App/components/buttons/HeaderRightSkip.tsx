import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

import { useTheme } from '../../contexts/theme'

interface HeaderButtonProps {
  title: string
  testID?: string
  accessibilityLabel?: string
  onPress?: () => void
  disabled?: boolean
}

const HeaderRight: React.FC<HeaderButtonProps> = ({ title, testID, accessibilityLabel, onPress, disabled = false }) => {
  const accessible = accessibilityLabel && accessibilityLabel !== '' ? true : false
  const { ColorPallet, TextTheme, Assets } = useTheme()
  const style = StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    touchableArea: {
      marginRight: 14,
    },
    title: {
      ...TextTheme.label,
      color: ColorPallet.brand.headerText,
      marginRight: 4,
    },
    icon: {
      transform: [{ rotate: '180deg' }],
    },
  })

  return (
    <TouchableOpacity
      accessible={accessible}
      accessibilityLabel={accessibilityLabel}
      testID={testID}
      onPress={onPress}
      style={[style.touchableArea]}
      disabled={disabled}
    >
      <View style={style.container}>
        <Text style={[style.title]}>{title}</Text>
        <Assets.svg.arrow
          height={TextTheme.label.fontSize}
          width={TextTheme.label.fontSize}
          fill={ColorPallet.brand.headerIcon}
          style={[style.icon]}
        />
      </View>
    </TouchableOpacity>
  )
}

export default HeaderRight
