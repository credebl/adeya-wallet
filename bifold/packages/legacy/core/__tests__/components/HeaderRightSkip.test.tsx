import { render } from '@testing-library/react-native'
import React from 'react'

import HeaderRightSkip from '../../App/components/buttons/HeaderRightSkip'

describe('Header Right Button Component', () => {
  test('Renders correctly', () => {
    const tree = render(
      <HeaderRightSkip
        title={'Hello Header'}
        testID={'TestID.Button'}
        accessibilityLabel={'Hello'}
        onPress={() => {
          return
        }}
      />
    )

    expect(tree).toMatchSnapshot()
  })
})
