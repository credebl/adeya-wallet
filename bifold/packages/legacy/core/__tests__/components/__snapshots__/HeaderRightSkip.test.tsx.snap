// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Right Button Component Renders correctly 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 14,
      "opacity": 1,
    }
  }
  testID="TestID.Button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginRight": 4,
          },
        ]
      }
    >
      Hello Header
    </Text>
    <
      fill="#FFFFFF"
      height={14}
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
          },
        ]
      }
      width={14}
    />
  </View>
</View>
`;
